src/promptflow-azure/promptflow/azure/operations/_async_run_uploader.py

```diff
@@ -180,9 +180,17 @@ async def _upload_snapshot(self) -> str:
         """Upload run snapshot to cloud. Return the cloud relative path of flow definition file."""
         logger.debug(f"Uploading snapshot for run {self.run.name!r}.")
         local_folder = self.run_output_path / LocalStorageFilenames.SNAPSHOT_FOLDER
-        # for some types of flex flow, even there is no flow.flex.yaml in the original flow folder,
-        # it will be generated in the snapshot folder in the .runs folder, so we can upload it to cloud as well.
-        _, flow_file = resolve_flow_path(local_folder)
+
+        # parse the flow definition file
+        run_flow_path = Path(self.run.properties[FlowRunProperties.FLOW_PATH]).name
+        if str(run_flow_path).endswith(PROMPTY_EXTENSION):
+            # for prompty flow run, get prompty file name from properties/flow_path
+            flow_file = run_flow_path
```

Comment:
`resolve_flow_path` is widely used and maybe we can do such update with a separate PR

Reply:
Good question. The answer is it could be, but I chose not to.
The reason is that, for current way, the parse logic will not rely on the original flow folder but the snapshot folder which is a more stable place (consider original flow folder can probably get moved to other places, in that situation the parse here will fail).

In fact, current local to cloud will not have the problem tho because it always upload right after the run is finished. But in the future this function may apply to existing run which was finished a while ago, and then the original flow file can be moved to other places. In this case, current way will not be affected but putting in resolve_flow_path will get impacted.